rdt/__init__.py

```diff
@@ -92,3 +93,81 @@ def get_demo(num_rows=5):
             'dollars_spent': dollars_spent
         })
     ], ignore_index=True)
+
+
+def _find_addons():
+    """Find and load add-ons based on the given group.
+
+    Args:
+        group (str):
+            The name of the entry points group to load.
+        parent_globals (dict):
+            The caller's global scope. Modules will be added to the parent's global scope through
+            their name.
+        add_all (bool):
+            Whether to also add everything in the add-on's ``module.__all__`` to the parent's
+            global scope. Defaults to ``False``.
+    """
+    group = 'rdt_modules'
+    for entry_point in iter_entry_points(group=group):
+        try:
+            addon = entry_point.load()
+        except Exception:  # pylint: disable=broad-exception-caught
+            msg = f'Failed to load "{entry_point.name}" from "{entry_point.module_name}".'
+            warnings.warn(msg)
+            continue
+
+        module_path, _, object_path = entry_point.name.partition(':')
+        module_path = module_path.split('.')
+
+        if module_path[0] != __name__:
+            msg = (
+                f"Failed to load '{entry_point.name}'. Expected base module to be '{__name__}'"
+                f", found '{module_path[0]}'."
+            )
+            warnings.warn(msg)
+            continue
+
+        base_module = modules[__name__]
```

Comment:
Could `modules` be `sys.modules` ? Took me a while to understand where this variable magically appears from.